File: src/main/java/com/yogesh/meditrack/service/DepartmentService.java
package com.yogesh.meditrack.service;

import com.yogesh.meditrack.entity.Department;

import java.util.List;

public interface DepartmentService {

    Department saveDepartment(Department department);

    List<Department> getAllDepartments();

    Department getDepartmentById(Long id);

    void deleteDepartment(Long id);
}

File: frontend/src/service/departmentService.js
import api from "./api";

export const getAllDepartments = () => api.get("/admin/departments");

export const getDepartmentById = (id) => api.get(`/admin/departments/${id}`);

export const saveDepartment = (dept) => api.post("/admin/departments", dept);

export const deleteDepartment = (id) => api.delete(`/admin/departments/${id}`);